### PR TITLE
[16.0][ADD] product_packaging_level_salable

### PR DESCRIPTION
**Pull request by QuocDuong1306**
This module feature was extracted from the original [sale-workflow/sale_by_packaging](https://github.com/oca/sale-workflow/tree/14.0/sale_by_packaging) module.

This module allows odoo users to set:
 - Can be sold: On product packaging type model, this checkbox defines if product
packagings from this particular type are available to be selected on sale
order line.

Depend on:
- https://github.com/OCA/product-attribute/pull/1215

### CODE REVIEW

File: product_packaging_level_sellable/models/sale_order_line.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        return self.product_packaging_id and not self.product_packaging_id.can_be_sold
+
+    @api.constrains("product_packaging_id")
+    def _check_product_packaging_can_be_sold(self):
+        for line in self:
+            if line._can_be_sold_error_condition():
+                raise ValidationError(
+                    _(
+                        "Packaging %(packaging)s on product %(product)s must be"
+                        " set as 'Can be sold' in order to be used on a sale order."
```
Comment (jbaudoux) [inline product_packaging_level_sellable/models/sale_order_line.py]:
```suggestion
                        " set as 'Sales' in order to be used on a sales order."
```

**Review by jbaudoux — CHANGES_REQUESTED** [review]
Can you add the migration scripts to move the old product packaging `can_be_sold` value to the new odoo `sales` field
And move the old packaging type "can_be_sold" to the new packaging level "can_be_sold"

**Comment by QuocDuong1306** [comment]
Updated changes request, migration script coming

**Comment by QuocDuong1306** [comment]
Hello @jbaudoux , i have updated the script for `move the old product packaging can_be_sold value to the new odoo sales field`

in this [PR](https://github.com/OCA/product-attribute/pull/1215), table `product_packaging_type` will be moved to `product_packaging_level` (including `can_be_sold` column), so we don't need to do anything particular here

**Review by marielejeune — APPROVED** [review]
LGTM (code review).
Now that https://github.com/OCA/product-attribute/pull/1215 is merged you can remove the commit modifying test-requirements.txt

**Comment by QuocDuong1306** [comment]
Thank @marielejeune , I updated

**Review by jbaudoux — APPROVED** [review]
Looks good but I would have named the module product_packaging_level_salable
to follow modern US English
https://www.merriam-webster.com/dictionary/salable

**Comment by QuocDuong1306** [comment]
@jbaudoux , I updated, could you help to have a look, thanks!

**Review by jbaudoux — APPROVED** [review]

**Pull request merged**